benchmark/layer_norm_fused.cpp

```diff
@@ -0,0 +1,208 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <device_lower/lower2device.h>
+#include <executor.h>
+#include <fusion.h>
+#include <ir_builder.h>
+#include <ir_utils.h>
+#include <ops/arith.h>
+#include <scheduler/all_schedulers.h>
+
+#include <benchmark/benchmark.h>
+
+#include <cuda_runtime.h>
+
+#include <benchmark/utils.h>
+#include <test/utils.h>
+
+using namespace nvfuser;
+
+//------------------------------------------------------------------------------
+
+static void setupLayerNormFused(Fusion* fusion, DataType dtype) {
+  TORCH_INTERNAL_ASSERT(dtype == DataType::Half);
+  const float kEps = 1e-5;
+
+  FusionGuard fg(fusion);
+  auto tv0 = makeContigTensor(1, dtype);
+  auto tv1 = makeContigTensor(2, dtype);
+  auto tv2 = makeContigTensor(1, dtype);
+  auto tv3 = makeContigTensor(1, dtype);
+  auto tv4 = makeContigTensor(1, dtype);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+  fusion->addInput(tv3);
+  fusion->addInput(tv4);
+  auto tv5 = broadcast(tv0, {true, false});
+  auto tv6 = castOp(DataType::Float, tv1);
+  auto tv7 = castOp(DataType::Float, tv5);
+  auto tv8 = add(tv6, tv7);
+  auto tv9 = castOp(DataType::Half, tv8);
+  auto tv10 = broadcast(tv2, {true, false});
+  auto tv11 = castOp(DataType::Float, tv9);
+  auto tv12 = castOp(DataType::Float, tv10);
+  auto tv13 = add(tv11, tv12);
+  auto tv14 = castOp(DataType::Half, tv13);
+  auto tv15 = castOp(DataType::Float, tv14);
+  auto tv16 = variance(tv15, {1}, false, false);
+  auto tv17 = broadcast(tv16, {false, true});
+  auto tv18 = sum(tv15, {1}, false);
+  auto tv19 = broadcast(tv18, {false, true});
+
+  nvfuser::Val* num_features =
+      IrBuilder::create<Double>(1, dtype = DataType::Double);
```

Comment:
What is this? I don't think this is C++ syntax.

Reply:
You're absolutely correct! This code was actually auto-translated from Python, hence the Pythonic style. Interestingly, it still functions correctly in C++ (looks like it just assign `DataType::Double` to `dtype` and then pass `dtype` to the function). As dtype isn't used after line 60, it doesn't impact the results. I'll submit a pull request to rectify this function call.